src/ejabberd_c2s.erl

```diff
@@ -1988,33 +1985,17 @@ presence_track(Acc, StateData) ->
             {Acc1, StateData#state{pres_i = I,
                                    pres_a = A}};
         <<"subscribe">> ->
-            Acc2 = mongoose_hooks:roster_out_subscription(
-                                           Server,
-                                           Acc,
-                                           User, To, subscribe),
-            Acc3 = check_privacy_and_route(Acc2, jid:to_bare(From), StateData),
-            {Acc3, StateData};
+            Acc1 = process_presence_subscription_and_route(Acc, subscribe, StateData),
+            {Acc1, StateData};
```

Comment:
How about returning the tuple from inside the `process_presence_subscribed_and_route` function? Do you think it's better to keep it outside?

Reply:
I just wanted to keep consistency with `check_privacy_and_route`, `check_privacy_and_route_or_ignore` and `check_privacy_and_route_probe` as they all seem to be pretty similar. But sure, this function could also return this tuple.

Reply:
Ok, fair enough. Thanks.